### PR TITLE
[SCB-736] generate default value to swagger for primitive type when there is no defaultValue annotation

### PR DESCRIPTION
**Pull request by maheshrajus**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/OperationGenerator.java [hunk 1]
```diff
@@ -315,8 +315,30 @@ protected void processByParameterAnnotation(Annotation[] paramAnnotations, int p
         }
       }
     }
-    if (parameter instanceof AbstractSerializableParameter && defaultValue != null) {
-      ((AbstractSerializableParameter<?>) parameter).setDefaultValue(defaultValue);
+    if (parameter instanceof AbstractSerializableParameter) {
+      if (defaultValue != null) {
+        ((AbstractSerializableParameter<?>) parameter).setDefaultValue(defaultValue);
+      } else if ((((AbstractSerializableParameter<?>) parameter).getDefaultValue() == null)
+          && (false == ((AbstractSerializableParameter<?>) parameter).getRequired())) { //if required false then only take java primitive values as defaults
```
Comment (liubao68) [inline swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/OperationGenerator.java]: use !

File: demo/demo-springmvc/springmvc-server/src/main/java/org/apache/servicecomb/demo/springmvc/server/SpringMvcDefaultValues.java [hunk 2]
```diff
@@ -64,4 +64,28 @@ public String query3(@RequestParam("a") @Min(value = 20) int a, @RequestParam("b
     return "Hello " + a + b;
   }
 
+  @PostMapping("/javaprimitive1")
+  public String springJavaPrim1(@RequestParam(name = "a") int a,
```
Comment (liubao68) [inline demo/demo-springmvc/springmvc-server/src/main/java/org/apache/servicecomb/demo/springmvc/server/SpringMvcDefaultValues.java]: From this demo and I made some tests, It seems for SpringMVC, there is a bug. 
For @RequestParam and other annotations, they got a required attribute, and default value is true. It seems that when generating swagger, we do not consider this attribute and it is always false. Event we write @RequestParam(name="a", requrired=true). Can you check that?

File: swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/OperationGenerator.java (hunk 1 above)
Comment (maheshrajus) [reply swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/OperationGenerator.java]: OK

File: demo/demo-springmvc/springmvc-server/src/main/java/org/apache/servicecomb/demo/springmvc/server/SpringMvcDefaultValues.java (hunk 2 above)
Comment (maheshrajus) [reply demo/demo-springmvc/springmvc-server/src/main/java/org/apache/servicecomb/demo/springmvc/server/SpringMvcDefaultValues.java]: yes,you are correct when we made @RequestParam(name="a", requrired=true) then also in swagger file it is showing required field is false. Looks to be defect in swagger generation module about springmvc.

Comment (maheshrajus) [reply demo/demo-springmvc/springmvc-server/src/main/java/org/apache/servicecomb/demo/springmvc/server/SpringMvcDefaultValues.java]: I fixed the base springmvc defect related to required parameter and raised PR. please check it.

Comment (liubao68) [reply demo/demo-springmvc/springmvc-server/src/main/java/org/apache/servicecomb/demo/springmvc/server/SpringMvcDefaultValues.java]: This default required is true, seems this test case will fail

Comment (maheshrajus) [reply demo/demo-springmvc/springmvc-server/src/main/java/org/apache/servicecomb/demo/springmvc/server/SpringMvcDefaultValues.java]: yes...i mofied in local evn but forgot to merge the demo. i am raising PR now. Thank you !

File: demo/demo-jaxrs/jaxrs-server/src/main/java/org/apache/servicecomb/demo/jaxrs/server/JaxRSDefaultValues.java [hunk 3]
```diff
@@ -88,4 +88,34 @@ public String queryPackages(HttpServletRequest httpRequest,
     return "" + appType;
   }
 
+  @Path("/javaprimitive1")
+  @POST
+  @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+  public String jaxRsJavaPrim1(@FormParam("a") int a, @DefaultValue("bobo") @FormParam("b") String b) {
```
Comment (wujimin) [inline demo/demo-jaxrs/jaxrs-server/src/main/java/org/apache/servicecomb/demo/jaxrs/server/JaxRSDefaultValues.java]: test case for different parameter type, do not named them 1/2/3/4......

Comment (maheshrajus) [reply demo/demo-jaxrs/jaxrs-server/src/main/java/org/apache/servicecomb/demo/jaxrs/server/JaxRSDefaultValues.java]: Fixed

**Pull request merged**